index.rst

```diff
@@ -1,61 +1,486 @@
-..
-  Technote content.
+:tocdepth: 1
 
-  See https://developer.lsst.io/restructuredtext/style.html
-  for a guide to reStructuredText writing.
+.. Please do not modify tocdepth; will be fixed when a new Sphinx theme is shipped.
 
-  Do not put the title, authors or other metadata in this document;
-  those are automatically added.
+.. sectnum::
 
-  Use the following syntax for sections:
+.. TODO: Delete the note below before merging new content to the master branch.
 
-  Sections
-  ========
+.. note::
 
-  and
+   **This technote is not yet published.**
 
-  Subsections
-  -----------
+   The butler stores datasets in collections along with software versions and pipeline configuration, but it does not store any information about the datasets that were used to generate a specific dataset. This document will discuss the plan for integrating provenance into Butler, sufficient to be able to regenerate any given dataset. It will refer to the requirements for provenance gathering specified in DMTN-185.
 
-  and
 
-  Subsubsections
-  ^^^^^^^^^^^^^^
+Existing concepts and mechanisms for provenance
+===============================================
 
-  To add images, add the image file (png, svg or jpeg preferred) to the
-  _static/ directory. The reST syntax for adding the image is
+Processing data with LSST middleware involves executing a directed acyclic graph (DAG) called a ``QuantumGraph``, where each node is either a ``Quantum``, a single execution of a single ``PipelineTask``, or a dataset managed by the ``Butler``.
```

Comment:
I am fine with the conceptual description and thought that might be what you were doing.  But I guess because of the wording it sounded like a description of the implementation.